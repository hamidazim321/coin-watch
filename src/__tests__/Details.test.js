import { render, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import fetchMock from 'jest-fetch-mock';
import DetailsPage from '../components/Details';
import { mockDetails } from './__mocks__/mockCoins';

fetchMock.enableMocks();

describe('DetailsPage', () => {

  beforeEach(() => {
    // Mock the fetch request to return the mockCoinData
    fetchMock.resetMocks();
    fetchMock.mockResponseOnce(JSON.stringify(mockDetails));
  });

  test('renders loading initially and then displays coin details', async () => {
    // Render the DetailsPage component with a mock route
    const { container, getByText } = render(
      <MemoryRouter initialEntries={['/Details/90']}>
        <DetailsPage />
      </MemoryRouter>
    );

    // Check if "Loading" text is displayed initially
    expect(getByText('Loading')).toBeInTheDocument();

    // Wait for the fetch request and data loading to complete
    await waitFor(() => {
      // Check if the coin details are displayed
      expect(getByText('BTC')).toBeInTheDocument();
      expect(getByText('Bitcoin')).toBeInTheDocument();
      expect(container).toMatchSnapshot()
    });
  });

  // Add more test cases as needed, e.g., test error handling, etc.

});
