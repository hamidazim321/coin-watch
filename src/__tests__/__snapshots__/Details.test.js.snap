// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders loading initially and then displays coin details 1`] = `
<div>
  <div
    class="details"
  >
    <div
      class="coin"
    >
      <div
        class="listHeader"
      >
        <p
          class="headline"
        >
          BTC
        </p>
        <span>
          Rank in Market Place:
          1
        </span>
      </div>
      <ul
        class="list"
      >
        <li>
          <p
            class="category"
          >
            Name
          </p>
          <span
            class="data"
          >
            Bitcoin
          </span>
        </li>
        <li>
          <p
            class="category"
          >
            Price
          </p>
          <span
            class="data"
          >
            $ 25767.96
          </span>
        </li>
        <li>
          <p
            class="category"
          >
            Market Cap
          </p>
          <span
            class="data"
          >
            $ 501726497779.56
          </span>
        </li>
        <li>
          <p
            class="category"
          >
            Price Change weekly
          </p>
          <span
            class="data"
          >
            -7.84
            %
          </span>
        </li>
        <li>
          <p
            class="category"
          >
            Price Change daily
          </p>
          <span
            class="data"
          >
            0.25
            %
          </span>
        </li>
        <li>
          <p
            class="category"
          >
            Price Change hourly
          </p>
          <span
            class="data"
          >
            0.06
            %
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
